ersilia/core/tracking.py

```diff
@@ -318,10 +342,10 @@ def check_types(self, result_df, metadata):
         count = 0
 
         # ignore key and input columns
-        dtypes_list = result_df.loc[:, ~result_df.columns.isin(["key", "input"])].dtypes
+        dtypes_list = {col: result[col] for col in result if col not in ["key", "input"]}
```

Comment:
When you do the defaultdict conversion here, you can skip these columns then instead of basically recreating a subset of the `columns` dict that is being created here.